rascal-lsp/src/main/rascal/lang/rascal/lsp/IDECheckerWrapper.rsc

```diff
@@ -0,0 +1,134 @@
+@license{
+Copyright (c) 2018-2025, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+module lang::rascal::lsp::IDECheckerWrapper
+
+import IO;
+import List;
+import Relation;
+import Set;
+import String;
+import analysis::graphs::Graph;
+import util::FileSystem;
+import util::Monitor;
+import util::Reflective;
+
+import lang::rascal::\syntax::Rascal;
+import lang::rascalcore::check::Checker;
+import lang::rascalcore::check::ModuleLocations;
+
+@synopsis{
+    This function is a wrapper for the Rascal type checker, taking into account that dependencies
+    may be open in the current workspace. It takes an overapproximation of the import/extend graph,
+    and calls the type checker on (transitive) dependencies first.
+
+    This function must only be used in an IDE context, as this is the only situation in which non-lib
+    source locations can occur in the `libs` entry of the PathConfig of a project. Note that for `lib`
+    locations, the type checker uses `tpl` files that are packaged with libraries.
+}
+list[ModuleMessages] checkFile(loc l, start[Module](loc file) getParseTree, PathConfig(loc file) getPathConfig) 
+    = job("Rascal check", list[ModuleMessages](void(str, int) step) {
+    checkForImports = [getParseTree(l)];
+    checkedForImports = {};
+    initialProject = inferProjectRoot(l);
+
+    rel[loc, loc] dependencies = {};
+    
+
+    step("Building dependency graph", 50);
```

Comment:
I know it's possible to grow the total size when we know about more modules. instead of guessing 50.

Reply:
For future ref: the answer is `jobTodo` .